### ISSUE TITLE
UAConfig.{accounts,entitlements} will cause tracebacks if we don't have {accounts,machine token} cached

### ISSUE THREAD

**Issue by OddBloke** [CanonicalLtd/ubuntu-advantage-client#205]
```py
def test_accounts():
    cfg = UAConfig({})
    cfg.accounts

def test_entitlements():
    cfg = UAConfig({})
    cfg.entitlements
```

gives

```pytb
======================================================================
ERROR: uaclient.tests.test_config.transplant_class.<locals>.C (test_accounts)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/daniel/dev/ubuntu-advantage-client/uaclient/tests/test_config.py", line 17, in test_accounts
    cfg.accounts
  File "/home/daniel/dev/ubuntu-advantage-client/uaclient/config.py", line 53, in accounts
    return self.read_cache('accounts')['accounts']
TypeError: 'NoneType' object is not subscriptable

======================================================================
ERROR: uaclient.tests.test_config.transplant_class.<locals>.C (test_entitlements)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/daniel/dev/ubuntu-advantage-client/uaclient/tests/test_config.py", line 17, in test_entitlements
    cfg.entitlements
  File "/home/daniel/dev/ubuntu-advantage-client/uaclient/config.py", line 95, in entitlements
    contractInfo = self.machine_token['machineTokenInfo']['contractInfo']
TypeError: 'NoneType' object is not subscriptable
```

For UAConfig.accounts, we should either coalesce the None out to the callers (all of which would need updating for that behaviour, as they all assume that there will be at least one account currently) or raise a more meaningful exception that can communicate something useful to the user.

For UAConfig.entitlements, we should consider the two above options, or return an empty dict (as it looks like most places that use it use `.get()` to handle an empty dict). (A meaningful exception is probably the best route.)

**Comment by blackboxsw** [comment]
Just validated that this is resolved in tip, likely due to fixing #196

**Comment by blackboxsw** [comment]
Adding the following diff to tip no longer results in tracebacks
diff --git a/uaclient/tests/test_config.py b/uaclient/tests/test_config.py
index e9648d5..2172b49 100644
--- a/uaclient/tests/test_config.py
+++ b/uaclient/tests/test_config.py
@@ -249,3 +249,10 @@ class TestDeleteCache(TestCase):
         assert [os.path.basename(t_file)] == os.listdir(tmp_dir)
         cfg.delete_cache()
         assert [os.path.basename(t_file)] == os.listdir(tmp_dir)
+
+    def test_1(self):
+        cfg = UAConfig({})
+        cfg.accounts
+        cfg.contracts
+        cfg.entitlements
+
